src/network/networkd-link.c

```diff
@@ -454,6 +454,24 @@ static int link_new(Manager *manager, sd_netlink_message *message, Link **ret) {
         if (r < 0)
                 return r;
 
+        r = hashmap_ensure_allocated(&manager->links_by_name, NULL);
+        if (r < 0)
+                return r;
+
+        r = hashmap_put(manager->links_by_name, link->ifname, link);
+        if (r < 0)
+                return r;
+
+        if (link->alternative_names) {
+                char **n;
+
+                STRV_FOREACH(n, link->alternative_names) {
+                        r = hashmap_put(manager->links_by_name, *n, link);
+                        if (r < 0)
+                                return r;
+                }
+        }
```

Comment:
the names need to be removed again, when the link goes away. And when the link is renamed/altnames added/removed the hash table also needs updating

Reply:
Ahh right. updated thanks.